File: CS/WinForms_MapControl_Lesson3/Form1.cs
using System;
using System.Data;
using System.Windows.Forms;
using DevExpress.XtraMap;

namespace WinForms_MapControl_Lesson3 {
    public partial class Form1 : Form {
        const String filepath = @"..\..\sochi2014.xml";
        VectorItemsLayer PieLayer { get { return (VectorItemsLayer)(mapControl1.Layers["PieLayer"]); } }
        PieChartDataAdapter PieAdapter { get { return (PieChartDataAdapter)PieLayer.Data; } }

        public Form1() {
            InitializeComponent();

            // Assign loaded data as data source for pie chart data adapter.
            PieAdapter.DataSource = LoadData(filepath);
        }

        // Loads data from XML.
        private DataTable LoadData(string path) {
            DataSet ds = new DataSet();
            ds.ReadXml(path);
            DataTable table = ds.Tables[0];
            return table;
        }
    }
}

File: VB/WinForms_MapControl_Lesson3/Form1.vb
Imports System
Imports System.Data
Imports System.Windows.Forms
Imports DevExpress.XtraMap

Namespace WinForms_MapControl_Lesson3
    Partial Public Class Form1
        Inherits Form

        Private Const filepath As String = "..\..\sochi2014.xml"
        Private ReadOnly Property PieLayer() As VectorItemsLayer
            Get
                Return CType(mapControl1.Layers("PieLayer"), VectorItemsLayer)
            End Get
        End Property
        Private ReadOnly Property PieAdapter() As PieChartDataAdapter
            Get
                Return CType(PieLayer.Data, PieChartDataAdapter)
            End Get
        End Property

        Public Sub New()
            InitializeComponent()

            ' Assign loaded data as data source for pie chart data adapter.
            PieAdapter.DataSource = LoadData(filepath)
        End Sub

        ' Loads data from XML.
        Private Function LoadData(ByVal path As String) As DataTable
            Dim ds As New DataSet()
            ds.ReadXml(path)
            Dim table As DataTable = ds.Tables(0)
            Return table
        End Function
    End Class
End Namespace
